book/src/concepts/entity_and_component.md

```diff
@@ -0,0 +1,105 @@
+
+## EntitiesRes
+
+Even though the structure of the entity is pretty simple, entity manipulation is very sophisticated and crucial to game performance. This is why entities are handled exclusively by the struct `EntitiesRes`. `EntitiesRes` provides two ways for creating/deleting entities:
+
+* Immediate creation/deletion, used for game setup or clean up.
+* Atomic or delayed creation/deletion, used in the game play state. It updates entities in batch at the end of each game loop.
+
+You will see how these methods are used in later chapters.
+
+
+## Declaring a component
+
+To declare a component, you first declare the relevant underlying data:
+
+```rust,no_run,noplaypen
+/// This `Component` describes the shape of an `Entity`
+enum Shape {
+    Sphere { radius: f32 },
+    RectangularPrism { height: f32, width: f32, depth: f32 },
+}
+
+/// This `Component` describes the transform of an `Entity`
+pub struct Transform {
+    position: Translation3<f32>,
+    rotation: UnitQuaternion<f32>,
+    scale: Vector3<f32>,
+}
+```
+
+and then you implement the `Component` trait for them:
+
+```rust,no_run,noplaypen
+# extern crate amethyst;
+# struct Shape;
+# struct Content;
+use amethyst::ecs::{Component, DenseVecStorage};
+
+impl Component for Shape {
+    type Storage = DenseVecStorage<Self>;
+}
+
+impl Component for Transform {
+    type Storage = FlaggedStorage<Self, DenseVecStorage<Self>>;
+}
+```
+The storage type will determine how you store the component, but it will not initiate the storage. Storage is initiated when you register a component in `World` or when you use that component in a System.
+
+## Storages
+
+There are a few storage strategies for different usage scenarios. The most commonly used types are `DenseVecStorage`, `VecStorage` and `FlaggedStorage`.
+
+* `DenseVecStorage`: Elements are stored in a contiguous vector. No empty space is left between `Component`s, allowing a lowered memory usage for big components.
+* `VecStorage`: Elements are stored into a sparse array. The entity id is the same as the index of component. If your component is small (<= 16 bytes) or is carried by most entities, this is preferable over `DenseVecStorage`.
+* `FlaggedStorage`: Used to keep track of changes of a component. Useful for caching purposes.
+
+![DenseVecStorage](../images/concepts/component_and_entity_densevecstorage.svg)
+
+![VecStorage](../images/concepts/component_and_entity_vecstorage.svg)
+
+For more information, see the [specs storage reference](https://docs.rs/specs/latest/specs/storage/index.html).
+
+There are a bunch more storages, and deciding which one is the best isn't trivial and should be done based on careful benchmarking. A general rule is: if your component is used in over 30% of entities, use VecStorage. If you don't know which one you should use, `DenseVecStorage` is a good default. It will need more memory than `VecStorage` for pointer-sized components, but it will perform well for most scenarios.
```

Comment:
Can we be confident enough to make the "over 30%" claim? (or did you find it somewhere online?)

I tend to leave it ambiguous instead of assigning numbers when I can't back up the claim (e.g. "If your component is commonly used" &ndash; it's up to interpretation what "common" means). Something else that can be said, "Benchmarking is the surefire way to determine which storage is optimal for your use case".

Reply:
It came from this comment by @jojolepro:
https://github.com/amethyst/amethyst/pull/1252#discussion_r241756263

Reply:
ah, I always hear this voice calling out "citation needed" when writing that kind of sentence 😛.
I'd trust @torkleyy's words over my thoughts =D